partner_communication_switzerland/models/account_invoice_line.py

```diff
@@ -75,11 +75,16 @@ def generate_thank_you(self):
         event = invoice_lines.mapped('event_id')
         ambassadors = invoice_lines.mapped('user_id')
 
+        event_id = event.id
+
+        if invoice_lines.mapped('gift_id'):
```

Comment:
Here you could look for contract instead of only gift, meaning that any donation related to a sponsorship is not considered as an event donation:
`if invoice_lines.mapped('contract_id'):`